compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/JMethodResolver.java

```diff
@@ -842,18 +842,35 @@ private String getParamTypesAsString(ParamTypeConstraint[] constraints) {
         return stringJoiner.toString();
     }
 
+    private JInteropException getNoSuchMethodError(String methodName, JMethod jMethod, BType bType,
+                                                   Class<?> declaringClass) {
+
+        Class<?>[] jParamTypes = jMethod.getParamTypes();
+        if (jParamTypes.length != 0) {
+            return new JInteropException(METHOD_SIGNATURE_DOES_NOT_MATCH,
+                    "Incompatible param type for method '" + methodName + "' in class '" + declaringClass.getName() +
+                            "': Java type '" + jParamTypes[0].getName() + "' will not be matched to ballerina type '" +
+                            (bType.tag == TypeTags.FINITE ? bType.tsymbol.name.value : bType) + "'");
+        } else {
+            return new JInteropException(METHOD_SIGNATURE_DOES_NOT_MATCH,
```

Comment:
Can we have some unit tests for this?
Like [this]( https://github.com/ballerina-platform/ballerina-lang/blob/0eccf22fc7943c87d7017b8c8e723e5612db224d/tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/javainterop/RefTypeNegativeTests.java#L39 ).